### PR TITLE
Throws exception when WaitContainerResultCallback timeouts

### LINKED ISSUE #570: WaitContainerResultCallback does not throw exception with information about timeout
Method `awaitStatusCode(long timeout, TimeUnit timeUnit)` does not throw exception with proper error message in case of a timeout. Now if waiting for a container timeouts,  `DockerClientException`  is thrown with generic message _"Error while wait container"_ (exception thrown from `getStatusCode()` method). This could be misleading.

### PR DESCRIPTION
**Pull request by orzeh**
Fixes #570

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/571)

### CODE REVIEW

File: src/test/java/com/github/dockerjava/core/command/WaitContainerCmdImplTest.java [hunk 1]
```diff
@@ -104,4 +102,31 @@ public void testWaitContainerAbort() throws Exception {
...
+        WaitContainerResultCallback callback = dockerClient.waitContainerCmd(container.getId()).exec(
+                new WaitContainerResultCallback());
+        try {
+            callback.awaitStatusCode(100, TimeUnit.MILLISECONDS);
+            fail("Should throw exception on timeout.");
+        } catch(DockerClientException e){
+            LOG.info(e.getMessage());
+        }
+
+        dockerClient.killContainerCmd(container.getId()).exec();
```
Comment (marcuslinke) [inline src/test/java/com/github/dockerjava/core/command/WaitContainerCmdImplTest.java]: Why you killing container here? Container created while test will be removed automatically after test.

**Comment by marcuslinke** [comment]
There is a test missing for the netty implementation. Tests are currently doubled for each implementation. Could you add it please?

File: src/test/java/com/github/dockerjava/core/command/WaitContainerCmdImplTest.java (hunk 1 above)
Comment (orzeh) [inline src/test/java/com/github/dockerjava/core/command/WaitContainerCmdImplTest.java]: I've just copied it from another test method `testWaitContainerAbort()`

Comment (marcuslinke) [inline src/test/java/com/github/dockerjava/core/command/WaitContainerCmdImplTest.java]: OK, understood. There it is part of the test. Please remove it in your test. Thanks.

Comment (marcuslinke) [inline src/test/java/com/github/dockerjava/core/command/WaitContainerCmdImplTest.java]: Hmmm. Seems `testWaitContainerAbort` is wrong but thats another story...

**Pull request merged**